cloud/amazon/iam_cert.py

```diff
@@ -0,0 +1,290 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+DOCUMENTATION = '''
+---
+module: iam_cert
+short_description: Manage server certificates for use on ELBs and CloudFront
+description:
+     - Allows for the management of server certificates
+version_added: "2.0"
+options:
+  name:
+    description:
+      - Name of certificate to add, update or remove.
+    required: true
+    aliases: []
+  new_name:
+    description:
+      - When present, this will update the name of the cert with the value passed here.
+    required: false
+    aliases: []
+  new_path:
+    description:
+      - When present, this will update the path of the cert with the value passed here.
+    required: false
+    aliases: []
+  state:
+    description:
+      - Whether to create, delete certificate. When present is specified it will attempt to make an update if new_path or new_name is specified.
+    required: true
+    default: null
+    choices: [ "present", "absent" ]
+    aliases: []
+  path:
+    description:
+      - When creating or updating, specify the desired path of the certificate
+    required: false
+    default: "/"
+    aliases: []
+  cert_chain:
+    description:
+      - The path to the CA certificate chain in PEM encoded format.
+    required: false
+    default: null
+    aliases: []
+  cert:
+    description:
+      - The path to the certificate body in PEM encoded format.
+    required: false
+    aliases: []
+  key:
+    description:
+      - The path to the private key of the certificate in PEM encoded format.
+  dup_ok:
+    description:
+      - By default the module will not upload a certifcate that is already uploaded into AWS. If set to True, it will upload the certifcate as long as the name is unique.
+    required: false
+    default: False
+    aliases: []
+  aws_secret_key:
+    description:
+      - AWS secret key. If not set then the value of the AWS_SECRET_KEY environment variable is used.
+    required: false
+    default: null
+    aliases: [ 'ec2_secret_key', 'secret_key' ]
+  aws_access_key:
+    description:
+      - AWS access key. If not set then the value of the AWS_ACCESS_KEY environment variable is used.
+    required: false
+    default: null
+    aliases: [ 'ec2_access_key', 'access_key' ]
+
+
+requirements: [ "boto" ]
+author: Jonathan I. Davila
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Basic server certificate upload
+tasks:
+- name: Upload Certifcate
+  iam_cert:
+    name: very_ssl
+    state: present
+    cert: somecert.pem
+    key: privcertkey
+    cert_chain: myverytrustedchain
+
+'''
+import json
+import sys
+try:
+    import boto
+    import boto.iam
+except ImportError:
+    print "failed=True msg='boto required for this module'"
+    sys.exit(1)
```

Comment:
Instead of calling sys.exit here you should call module.fail_json() and let Ansible handle the failure cleanly.  See [here](https://github.com/ansible/ansible-modules-core/blob/devel/cloud/amazon/ec2.py#L500-L507) for an example.